File: Desktop/GB/backend/models/feeding.py
from database import db
from datetime import datetime
from sqlalchemy import ForeignKey

class Feeding(db.Model):
    __tablename__ = 'feeding'
    
    id = db.Column(db.Integer, primary_key=True)
    cattle_id = db.Column(db.Integer, ForeignKey('cattle.id'), nullable=False)
    date_recorded = db.Column(db.Date, default=datetime.utcnow, nullable=False)
    feed_type = db.Column(db.String(100), nullable=False)  # Hay, Grain, Silage, etc.
    quantity_kg = db.Column(db.Float, nullable=False)
    cost_per_unit = db.Column(db.Float, nullable=True)
    total_cost = db.Column(db.Float, nullable=True)
    supplier = db.Column(db.String(100), nullable=True)
    notes = db.Column(db.Text, nullable=True)
    
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    def __repr__(self):
        return f'Feeding(id={self.id}, cattle_id={self.cattle_id}, feed_type={self.feed_type}, qty={self.quantity_kg})'

    def to_dict(self):
        return {
            'id': self.id,
            'cattle_id': self.cattle_id,
            'date_recorded': self.date_recorded.isoformat(),
            'feed_type': self.feed_type,
            'quantity_kg': self.quantity_kg,
            'cost_per_unit': self.cost_per_unit,
            'total_cost': self.total_cost,
            'supplier': self.supplier,
            'notes': self.notes,
            'created_at': self.created_at.isoformat(),
            'updated_at': self.updated_at.isoformat()
        }

File: Desktop/GB/cattle-management-mobile/backend-mongo/models/Feeding.js
const mongoose = require('mongoose');

const feedingSchema = new mongoose.Schema({
  cattle_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Cattle',
    required: true
  },
  date_recorded: {
    type: Date,
    required: true,
    default: Date.now
  },
  feed_type: {
    type: String,
    required: true,
    enum: ['Hay', 'Corn Silage', 'Barley', 'Wheat', 'Alfalfa', 'Grass Pellets', 'Protein Supplement', 'Other']
  },
  quantity_kg: {
    type: Number,
    required: true,
    min: 0
  },
  cost_per_unit: {
    type: Number,
    min: 0
  },
  total_cost: {
    type: Number,
    min: 0
  },
  supplier: {
    type: String,
    trim: true
  },
  notes: {
    type: String,
    trim: true
  }
}, {
  timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' }
});

// Pre-save middleware to calculate total_cost
feedingSchema.pre('save', function(next) {
  if (this.cost_per_unit && this.quantity_kg) {
    this.total_cost = this.cost_per_unit * this.quantity_kg;
  }
  next();
});

// Indexes
feedingSchema.index({ cattle_id: 1 });
feedingSchema.index({ date_recorded: -1 });
feedingSchema.index({ feed_type: 1 });

module.exports = mongoose.model('Feeding', feedingSchema);
